Tests/OpenCombineTests/PublisherTests/DecodeTests.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  DecodeTests.swift
+//  
+//
+//  Created by Joseph Spadafora on 6/21/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class DecodeTests: XCTestCase {
+    static let allTests = [
+        ("testDecodeWorks", testDecodeWorks),
+        ("testDownstraemReceivesFailure", testDownstreamReceivesFailure),
+        ("testDemand", testDemand)
+    ]
+
+    private let jsonEncoder = JSONEncoder()
+    private let jsonDecoder = JSONDecoder()
+
+    func testDecodeWorks() throws {
+        // Given
+        let data = try jsonEncoder.encode(testValue)
+        let subject = PassthroughSubject<Data, Error>()
+        let publisher = subject.decode(type: [String: String].self, decoder: jsonDecoder)
+        let subscriber = TrackingSubscriberBase<[String: String], Error>()
+
+        // When
+        publisher.subscribe(subscriber)
+        subject.send(data)
+
+        // Then
+        XCTAssertEqual(subscriber.history, [.subscription(Subscriptions.empty),
+                                            .value(testValue)])
+    }
+
+    func testDownstreamReceivesFailure() {
+        // Given
+        let failData = Data("whoops".utf8)
+        let subject = PassthroughSubject<Data, Error>()
+        let publisher = subject.decode(type: [String: String].self, decoder: jsonDecoder)
+        let subscriber = TrackingSubscriberBase<[String: String], Error>()
+
+        // When
+        publisher.subscribe(subscriber)
+        subject.send(failData)
+
+        // Then
+        guard case .completion(.failure) = subscriber.history[1] else {
+            XCTFail("Decode failure not found")
+            return
+        }
```

Comment:
I have this test crashing in compatibility mode because of index out of bounds error.

Let's compare the whole history for equality:

```swift
// Then
XCTAssertEqual(subscriber.history, [.subscription(Subscriptions.empty), .completion(.failure)])
```

Reply:
The JSONDecoder is throwing the error here which is why I'm using the `guard case` syntax so that the error comparison doesn't get messy here.

Reply:
Looks like any decoding error compares to equal here, so this should work